### PR TITLE
Change default return type for integer values to BigInt

### LINKED ISSUE #5128: Change default return type for integer values to BigInt
### This is not a Bug Report, Feature Request, or related to Documentation

- [X] I have searched the existing issues

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What's up?

Based on the [discussion](https://github.com/ChainSafe/web3.js/discussions/5120). It seems to better to use the `BigInt` as the default return type for the web3. 

### Examples/References

_No response_

### PR DESCRIPTION
**Pull request by nazarhussain**
## Description

Fixes #5128

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` with success.
- [x] I have tested the built `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

**Review by nikoulai — APPROVED** [review]

**Review by avkos — APPROVED** [review]

**Review by avkos — CHANGES_REQUESTED** [review]
Some of the tests are not passing.  it will be great to fix it

File: packages/web3-common/src/formatter.ts [hunk 1]
```diff
@@ -60,9 +60,9 @@ export type DataFormat = {
 	readonly bytes: FMT_BYTES;
 };
 
-export const DEFAULT_RETURN_FORMAT = { number: FMT_NUMBER.HEX, bytes: FMT_BYTES.HEX } as const;
+export const DEFAULT_RETURN_FORMAT = { number: FMT_NUMBER.BIGINT, bytes: FMT_BYTES.HEX } as const;
+export const ETH_DATA_FORMAT = { number: FMT_NUMBER.HEX, bytes: FMT_BYTES.HEX } as const;
```
Comment (jdevcs) [inline packages/web3-common/src/formatter.ts:64]:
```suggestion
export const ETH_INPUT_DATA_FORMAT = { number: FMT_NUMBER.HEX, bytes: FMT_BYTES.HEX } as const;
```
I think should rename its all occurrences for better readability.

Comment (jdevcs) [inline packages/web3-common/src/formatter.ts:64]:
```
export const DEFAULT_RETURN_FORMAT = { number: FMT_NUMBER.BIGINT, bytes: FMT_BYTES.HEX } as const;
export const ETH_DATA_FORMAT = { number: FMT_NUMBER.HEX, bytes: FMT_BYTES.HEX } as co
nst;
```

These should be places in common/types.ts

**Review by jdevcs — DISMISSED** [review]

File: packages/web3-eth-ens/src/ens.ts [hunk 2]
```diff
@@ -289,7 +292,10 @@ export class ENS extends Web3Context<EthExecutionAPI & Web3NetAPI> {
 		if (this._detectedAddress) {
 			return this._detectedAddress;
 		}
-		const networkType = await getId(this, DEFAULT_RETURN_FORMAT); // get the network from provider
+		const networkType = await getId(this, {
+			...DEFAULT_RETURN_FORMAT,
+			number: FMT_NUMBER.HEX,
```
Comment (jdevcs) [inline packages/web3-eth-ens/src/ens.ts:297]: If we always want to have `number:FMT_NUMBER.HEX` here why we are using default format?

File: packages/web3-eth/src/constants.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { FMT_BYTES, FMT_NUMBER } from 'web3-common';
+
+export const HEX_NUMBER_DATA_FORMAT = { bytes: FMT_BYTES.HEX, number: FMT_NUMBER.HEX } as const;
```
Comment (jdevcs) [inline packages/web3-eth/src/constants.ts]: `HEX_NUMBER_DATA_FORMAT ` is only used in send signed transaction tests so should be places in tests dir.

**Review by avkos — APPROVED** [review]

File: packages/web3-eth/test/integration/web3_eth/send_signed_transaction.test.ts [hunk 4]
```diff
@@ -59,15 +65,25 @@ describe('Web3Eth.sendSignedTransaction', () => {
 				...transaction,
 				...gasPricing,
 			});
-			const response = await web3Eth.sendSignedTransaction(signedTransaction.raw);
-			expect(response.status).toBe('0x1');
+			const response = await web3Eth.sendSignedTransaction(
+				typeof signedTransaction.raw === 'string'
+					? signedTransaction.raw
+					: signedTransaction.raw.raw,
```
Comment (jdevcs) [inline packages/web3-eth/test/integration/web3_eth/send_signed_transaction.test.ts:78]: There shouldn't be nested raw for output of `web3Eth.signTransaction`. I think it has to do with recently discussed geth vs ganache output diff?

Comment (nazarhussain) [reply packages/web3-eth/test/integration/web3_eth/send_signed_transaction.test.ts:78]: Yes this change didn't appear from this PR. It was done earlier in @spacesailor24 PR and relates to the output difference for ganache and geth.

File: packages/web3-validator/src/validation/numbers.ts [hunk 5]
```diff
@@ -36,6 +36,10 @@ export const isUInt = (
 		return false;
 	}
 
+	if (typeof value === 'string' && value.length === 0) {
```
Comment (jdevcs) [inline packages/web3-validator/src/validation/numbers.ts]: Just remove line 35,36, and 37 above and can use:
```suggestion
	if ( (!['number', 'string', 'bigint'].includes(typeof value))  ||
	    (typeof value === 'string' && value.length === 0) ) {
```

File: packages/web3-eth/test/integration/web3_eth/send_signed_transaction.test.ts (hunk 4 above)
Comment (jdevcs) [reply packages/web3-eth/test/integration/web3_eth/send_signed_transaction.test.ts:78]: ok, https://github.com/ChainSafe/web3.js/issues/5077 with this issue we will need to fix these tests as well.

File: packages/web3-common/src/formatter.ts (hunk 1 above)
Comment (nazarhussain) [reply packages/web3-common/src/formatter.ts:64]: It's value not the type.

Comment (nazarhussain) [reply packages/web3-common/src/formatter.ts:64]: @jdevcs As per my knowledge, Eth implementation follows the same format for input/output. In that aspect I named this value as `ETH_DATA_FORMAT`. Please suggest if you think `ETH_INPUT_DATA_FORMAT` is better usage?

File: packages/web3-eth-ens/src/ens.ts (hunk 2 above)
Comment (nazarhussain) [reply packages/web3-eth-ens/src/ens.ts:297]: To not redefine `bytes` format. You can't define just one format while declaring one data format.

**Comment by nazarhussain** [comment]
@jdevcs Need to merge this PR as another PR is based on it. If you think we still need to make further changes, we can do it with a separate issue.

**Pull request merged**